### PR TITLE
Rpo2023 challenge storing groceries

### PR DESCRIPTION
**Pull request by P-ict0**

### CODE REVIEW

File: challenge_storing_groceries/src/challenge_storing_groceries/inspect_shelves.py [hunk 1]
```diff
@@ -102,3 +104,18 @@ def __init__(self, robot, entityDes, objectIDsDes=None, searchAreas=None, naviga
                                                   unknown_threshold=unknown_threshold,
                                                   filter_threshold=filter_threshold),
                                    transitions={'done': 'ITERATE_AREA'})
+
+
+if __name__ == '__main__':
```
Comment (MatthijsBurgh) [inline challenge_storing_groceries/src/challenge_storing_groceries/inspect_shelves.py:109]:
```suggestion
if __name__ == "__main__":
```

File: challenge_storing_groceries/src/challenge_storing_groceries/manipulate_machine.py [hunk 2]
```diff
@@ -60,6 +62,18 @@ def __init__(self, robot, item_designator, place_pose_designator, arm=None):
...
+
+            smach.StateMachine.add("SAY_HANDOVER",
+                                   states.human_interaction.Say(robot, ["I failed to place the object, "
+                                                                        "I will hand it over to you"]),
+                                   transitions={'spoken': 'HANDOVER'}
+                                   )
+
+            smach.StateMachine.add("HANDOVER",
+                                   states.manipulation.HandoverToHuman(robot, arm),
+                                   transitions={'succeeded': 'succeeded',
```
Comment (MatthijsBurgh) [inline challenge_storing_groceries/src/challenge_storing_groceries/manipulate_machine.py:76]: Not sure this should result in succeeded. Probably failed is better.

File: challenge_storing_groceries/src/challenge_storing_groceries/manipulate_machine.py [hunk 3]
```diff
@@ -132,13 +169,54 @@ def check_similar_item(userdata=None):
...
+    robot.reset_all_arms()
+
+    entityDes = EntityByIdDesignator(robot, uuid=sys.argv[2])
+
+    shelfDes = EntityByIdDesignator(robot, uuid="cabinet")
+    armDes = ArmDesignator(robot)
+    placePoseDes = EmptySpotDesignator(robot, shelfDes, armDes, area=place_area, name="empty_spot_designator")
+
+    sm = StoreSingleItem(robot, entityDes, placePoseDes)
+    sm.execute()
```
Comment (PetervDooren) [inline challenge_storing_groceries/src/challenge_storing_groceries/manipulate_machine.py:222]: double check names and usage string. Or remove here and fix in https://github.com/tue-robotics/tue_robocup/pull/1270

File: challenge_storing_groceries/src/challenge_storing_groceries/near_object_designator.py [hunk 4]
```diff
@@ -180,3 +176,23 @@ def _determine_points_of_interest(self, near_entity, radius, spacing):
...
+
+    rospy.init_node('test_inspect_shelves')
+
+    robot = get_robot("hero")
+
+    shelfDes = EntityByIdDesignator(robot, uuid="closet")
+    entityDes = EntityByIdDesignator(robot, uuid=sys.argv[1])
+
+    nearObjectDes = NearObjectSpotDesignator(robot, entityDes, shelfDes)
+    nearObjectDes.resolve()
```
Comment (PetervDooren) [inline challenge_storing_groceries/src/challenge_storing_groceries/near_object_designator.py:198]: double check names and usage string. Or remove here and fix in https://github.com/tue-robotics/tue_robocup/pull/1270

File: challenge_storing_groceries/src/challenge_storing_groceries/open_door.py [hunk 5]
```diff
@@ -6,12 +6,10 @@
 import rospy
 import smach
 import tf2_ros
+# noinspection PyUnresolvedReferences
```
Comment (PetervDooren) [inline challenge_storing_groceries/src/challenge_storing_groceries/open_door.py:9]: Do we recall the reason for this? If not lets remove this.

**Review by PetervDooren — CHANGES_REQUESTED** [review]
Few clean up things. Most are related to testcases which can be defered to another PR. Other than that, we should test this once in Impuls and merge it.

File: robocup_knowledge/src/robocup_knowledge/environments/impuls/challenge_storing_groceries.py [hunk 6]
```diff
@@ -9,7 +9,11 @@
 
 # Detection
 shelf = "closet"
-object_types = [obj["name"] for obj in common.objects]
+default_area = "shelf4"
+inspect_area = "in_front_of"
+object_shelves = ["shelf3", "shelf4", "shelf5"] #TODO unused variable?
+object_types = [obj["name"] for obj in common.objects] #TODO unused variable?
```
Comment (MatthijsBurgh) [inline robocup_knowledge/src/robocup_knowledge/environments/impuls/challenge_storing_groceries.py:15]:
```suggestion
object_shelves = ["shelf3", "shelf4", "shelf5"]  # TODO unused variable?
object_types = [obj["name"] for obj in common.objects]  # TODO unused variable?
```

File: challenge_storing_groceries/src/challenge_storing_groceries/open_door.py (hunk 5 above)
Comment (PetervDooren) [reply challenge_storing_groceries/src/challenge_storing_groceries/open_door.py:9]: open door needs a lot of work (probably a complete refactor) anyway. So I am fine either way.

**Review by PetervDooren — APPROVED** [review]

**Comment by MatthijsBurgh** [comment]
@PetervDooren changes in `robot_smach_states` should be applied to the remainder branch. This has created merge conflicts

**Pull request merged**